tests/vm_migration_test.go

```diff
@@ -89,6 +89,52 @@ var _ = Describe("VmMigration", func() {
 			close(done)
 		}, 30)
 
+		It("Should respect and preserve pre-set node affinity on the VM", func(done Done) {
+			// Prepare dummy affinity rule
+			sourceVM.Spec.Affinity = &v1.Affinity{
+				NodeAffinity: &k8sv1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+						NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+							{
+								MatchExpressions: []k8sv1.NodeSelectorRequirement{
+									{
+										Key:      "invalidtag",
+										Values:   []string{"nothing"},
+										Operator: k8sv1.NodeSelectorOpNotIn,
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			vm, err := virtClient.RestClient().Post().Resource("vms").Namespace(tests.NamespaceTestDefault).Body(sourceVM).Do().Get()
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMStart(vm)
+
+			migration := tests.NewRandomMigrationForVm(sourceVM)
+			err = virtClient.RestClient().Post().Resource("migrations").Namespace(tests.NamespaceTestDefault).Body(migration).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() v1.MigrationPhase {
+				obj, err := virtClient.RestClient().Get().Resource("migrations").Namespace(tests.NamespaceTestDefault).Name(migration.ObjectMeta.Name).Do().Get()
+				Expect(err).ToNot(HaveOccurred())
+				var m *v1.Migration = obj.(*v1.Migration)
+				return m.Status.Phase
+			}, TIMEOUT, POLLING_INTERVAL).Should(Equal(v1.MigrationSucceeded))
```

Comment:
The test log shows that migration was not done after the timeout. The test looks like this should work. Did you try it yourself already?

Reply:
It timeouts for me locally and it worked in CI when I increased the global timeout to 30s